mqtt-streaming/src/main/scala/akka/stream/alpakka/mqtt/streaming/impl/ClientState.scala

```diff
@@ -226,8 +257,13 @@ import scala.util.{Failure, Success}
               )
             )
           } else {
+            data.activeProducers.values.foreach { producer =>
+              producer ! Producer.ReceiveConnect
+            }
```

Comment:
Main change on the client to let the producers know about new `ReceiveConnect` so that they can republish.